File: ui-testing/pages/checkout_complete.py
from pages.base_page import BasePage
from selenium.webdriver.common.by import By


class CheckoutComplete(BasePage):
    COMPLETE_TITLE = (By.CLASS_NAME, "title")
    COMPLETE_HEADER = (By.CLASS_NAME, "complete-header")
    COMPLETE_TEXT = (By.CLASS_NAME, "complete-text")
    BACK_HOME_BTN = (By.ID, 'back-to-products')

    def get_complete_title(self):
        # Checkout: Complete!
        return self.get_text(self.COMPLETE_TITLE)

    def get_complete_header(self):
        # THANK YOU FOR YOUR ORDER
        return self.get_text(self.COMPLETE_HEADER)

    def get_complete_text(self):
        # Your order has been dispatched, and will arrive just as fast as the pony can get there!
        return self.get_text(self.COMPLETE_TEXT)

    def click_back_home_btn(self):
        self.click(self.BACK_HOME_BTN)

File: playwright-testing/pages/checkout_complete.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './base.page';


export class CheckoutComplete extends BasePage{
    readonly completeTitle : Locator;
    readonly completeHeader : Locator;    
    readonly completeText : Locator;
    readonly backHomeBtn : Locator;
    

    constructor(page : Page){
        super(page);
        this.completeTitle = page.locator('.title');
        this.completeHeader = page.locator('.complete-header');
        this.completeText = page.locator('.complete-text');
        this.backHomeBtn = page.locator('#back-to-products')
    }
    async getCompleteTitle(): Promise<string> {
        return await this.completeTitle.innerText();
    }
    async getCompleteHeader(): Promise<string> {
        return await this.completeHeader.innerText();
    }
    async getCompleteText(): Promise<string> {
        return await this.completeText.innerText();
    }   

    async clickBackHomeBtn(): Promise<void> {
        await this.backHomeBtn.click();
    }




}